--- icparc_solvers/Makefile.x86_64_linux.orig	2014-07-08 01:21:56.580387319 +0200
+++ icparc_solvers/Makefile.x86_64_linux	2014-07-08 01:24:49.220390179 +0200
@@ -30,7 +30,7 @@
 ARCH = x86_64_linux
 top_srcdir = ..
 PREFIX = /var/tmp/portage/dev-lang/eclipse-clp-6.1.189/image
-ECLIPSEDIR = $(PREFIX)
+ECLIPSEDIR = $(top_srcdir)
 MAKEFILE = Makefile.$(ARCH)
 OSCLASS = unix
 
@@ -47,9 +47,9 @@
 GMP_LD = 
 FLOAT_ROUND_FLAGS = 
 
-CPPFLAGS  = -I. -I$(ECLIPSEDIR)/include/$(ARCH) -I$(ECLIPSEDIR)/Kernel/src -I- $(OS_INCLUDES)
+CPPFLAGS  = -I. -I$(PREFIX)/include/$(ARCH) -I$(ECLIPSEDIR)/Kernel/src -I- $(OS_INCLUDES)
 STDCXXLIB = -Wl,-Bstatic -lstdc++ -Wl,-Bdynamic
-ECLIBS	= -L$(ECLIPSEDIR)/lib/$(ARCH) -leclipse -L.  -lgmp -lpthread -lnsl -ldl -lm 
+ECLIBS	= -L$(PREFIX)/lib/$(ARCH) -leclipse -L.  -lgmp -lpthread -lnsl -ldl -lm 
 
 
 PERM = a+r,u+w,go-w
@@ -248,7 +248,7 @@
 # Installation
 #----------------------------------------------------------------------
 
-ECLIPSE = $(ECLIPSEDIR)/bin/$(ARCH)/eclipse
+ECLIPSE = $(PREFIX)/bin/$(ARCH)/eclipse
 
 EXAMPLE_FILES =	cbs.ecl lds.ecl wcs.ecl knapsack_ls.ecl
 ECL_ECO_FILES =	graph_algorithms.ecl heap_array.ecl \
