--- Eclipse_6.1_187/Usc/src/configure
+++ Eclipse_6.1_187/Usc/src/configure
@@ -526,8 +526,9 @@
 #
 # UNIX variants
 #
-CC=cc
-CFLAGS="-O"
+CC=${CC:-cc}
+AR=${AR:-ar}
+CFLAGS=${CFLAGS:-"-O"}
 case "$ARCH" in
  sun3)
     ;;
@@ -587,7 +588,7 @@
 EOF
 
     ;;
- i386_linux)
+ i386_linux|x86_64_linux)
     LIBS="-lieee $LIBS"
     ;;
  rs_aix32)
@@ -950,6 +951,7 @@
 s%@mandir@%$mandir%g
 s%@CPP@%$CPP%g
 s%@CC@%$CC%g
+s%@AR@%$AR%g
 s%@DFLAGS@%$DFLAGS%g
 s%@ARCH@%$ARCH%g
 s%@RANLIB@%$RANLIB%g
--- Eclipse_6.1_187/Usc/src/Makefile.in
+++ Eclipse_6.1_187/Usc/src/Makefile.in
@@ -18,7 +18,7 @@
 
 CCFLAGS = $(CFLAGS) $(DFLAGS)
 
-AR = ar ruv
+AR = @AR@ ruv
 RM = /bin/rm -f
 CP = /bin/cp
 
@@ -28,8 +28,7 @@
 .c.o:
 	$(CC) $(CCFLAGS) -c $*.c 
 
-all:	
-	$(MAKE) libusc.a
+all: libusc.a
 
 install: all
 	-$(RM) ../include/usc.h
