--- a/setup.py	2019-06-10 16:27:47.000000000 +0200
+++ b/setup.py.lala	2019-09-06 13:50:47.079955775 +0200
@@ -43,7 +43,7 @@
     zip_safe=False,
     platforms='any',
     install_requires=["django>1.11", "mongoengine>=0.14"],
-    packages=find_packages(exclude=('doc', 'docs',)),
+    packages=find_packages(exclude=('doc', 'docs','test*')),
     include_package_data=True,
     # use python setup.py nosetests to test
     setup_requires=['nose', 'coverage'],
