--- codegen/pygobject-codegen-2.0.in
+++ codegen/pygobject-codegen-2.0.in
@@ -4,7 +4,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 datarootdir=@datarootdir@
 datadir=@datadir@
-codegendir=@pyexecdir@/gtk-2.0/codegen
+codegendir=$(python2 -c "from distutils import sysconfig; print sysconfig.get_python_lib(1, 0, prefix='@exec_prefix@')")/gtk-2.0/codegen
 
 PYTHONPATH=$codegendir
 export PYTHONPATH
