--- xbmc-9999/configure.in.orig	2012-06-18 20:01:43.283968125 +0200
+++ xbmc-9999/configure.in	2012-06-18 20:03:30.525720533 +0200
@@ -575,25 +575,24 @@
 case $use_platform in
   raspberry-pi)
      target_platform=target_raspberry_pi
-     use_neon=no
+     use_neon="no"
      use_texturepacker=yes
-     use_texturepacker_native=yes
      use_arch="arm"
      use_cpu=arm1176jzf-s
-     use_armv5te="yes"
+     use_armv5te="no"
      use_armv6t2="no"
      use_armv6="yes"
      use_armvfp="yes"
      use_hardcoded_tables="yes"
-     use_alsa="no"
-     USE_TEXTUREPACKER_NATIVE_ROOT="$TEXTUREPACKER_NATIVE_ROOT"
      ARCH="arm"
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_ARMEL -DTARGET_RASPBERRY_PI")
      AC_DEFINE(HAS_EGLGLES, [1], [Define if supporting EGL based GLES Framebuffer])
      USE_OMXPLAYER=1; AC_DEFINE([HAVE_OMXPLAYER],[1],["Define to 1 if OMX Player is enabled"])
      USE_OMXLIB=1; AC_DEFINE([HAVE_OMXLIB],[1],["Define to 1 if OMX libs is enabled"])
-     CFLAGS="$CFLAGS"
-     CXXFLAGS="$CXXFLAGS"
+     CFLAGS="$CFLAGS -Wno-deprecated"
+     CXXFLAGS="$CXXFLAGS -Wno-deprecated"
+     CPPFLAGS="-I/opt/vc/include $CPPFLAGS"
+     LDFLAGS="-L/opt/vc/lib $LDFLAGS"
      ;;
   *)
      use_alsa="yes"
@@ -746,10 +745,11 @@
     AC_MSG_RESULT(== WARNING: OpenGLES support is assumed.)
   else
     if test "$target_platform" = "target_raspberry_pi"; then
-      AC_DEFINE([HAVE_LIBEGL],[1],["Define to 1 if you have the `EGL' library (-lEGL)."])
-      AC_DEFINE([HAVE_LIBGLESV2],[1],["Define to 1 if you have the `GLESv2' library (-lGLESv2)."])
-      AC_MSG_RESULT(== WARNING: OpenGLES support is assumed.)
-      LIBS="$LIBS -lEGL -lGLESv2 -lbcm_host -lvcos -lvchiq_arm"
+      AC_CHECK_LIB([EGL],   [main],, AC_MSG_ERROR($missing_library))
+      AC_CHECK_LIB([GLESv2],[main],, AC_MSG_ERROR($missing_library))
+      AC_CHECK_HEADER([EGL/egl.h],, AC_MSG_ERROR($missing_library))
+      AC_CHECK_HEADER([GLES2/gl2.h],, AC_MSG_ERROR($missing_library))
+      LIBS="$LIBS -lbcm_host -lvcos -lvchiq_arm"
     else
       AC_CHECK_LIB([EGL],   [main],, AC_MSG_ERROR($missing_library))
       AC_CHECK_LIB([GLESv2],[main],, AC_MSG_ERROR($missing_library))
