diff -Naur flask-mongoengine-0.7.2.orig/setup.py flask-mongoengine-0.7.2/setup.py
--- flask-mongoengine-0.7.2.orig/setup.py	2015-10-15 09:54:43.630006422 +0200
+++ flask-mongoengine-0.7.2/setup.py	2015-10-15 10:05:03.720008042 +0200
@@ -43,7 +43,7 @@
               'flask_mongoengine.wtf'],
     include_package_data=True,
     tests_require=test_requirements,
-    setup_requires=test_requirements,  # Allow proper nose usage with setuptools and tox
+    #setup_requires=test_requirements,  # Allow proper nose usage with setuptools and tox
     classifiers=[
         'Development Status :: 4 - Beta',
         'Environment :: Web Environment',
