--- flask-mongoengine-0.7.4.orig/setup.py	2015-12-04 17:29:10.780074994 +0100
+++ flask-mongoengine-0.7.4/setup.py	2015-12-04 17:29:46.270075087 +0100
@@ -58,7 +58,7 @@
               'flask_mongoengine.wtf'],
     include_package_data=True,
     tests_require=test_requirements,
-    setup_requires=test_requirements,  # Allow proper nose usage with setuptools and tox
+    #setup_requires=test_requirements,  # Allow proper nose usage with setuptools and tox
     description=DESCRIPTION,
     long_description=LONG_DESCRIPTION,
     classifiers=[
